app/views/posts/index.html.erb

```diff
@@ -4,6 +4,7 @@
   <p><%= simple_format post.message %></p>
   <%= post.created_at.strftime("%d %B %Y %H:%M") %>
   <p><%= link_to 'Delete', [current_user, post], :method => :delete  %></p>
+  <p><%= link_to 'Edit', [current_user, post]  %></p>
```

Comment:
<%= link_to "Edit", edit_user_post(current_user, post) :method => :edit %>

Try this instead of the above syntax, this way it should be automatically routed to the edit method rather than the show :)

Reply:
This has been changed. Merci beaucoup @LucyMHall de votre aide 🥖